Vb/countcard

vickycore opened the pull request.

vickycore committed:
Created CountCard component.

Refs #61 #62
Fixes #60

Signed-off-by: Vikram Bala <vikrambala.va@gmail.com>

vickycore committed:
Updated CountCard component to fix sizing, added Profile Error card to results.vue, created ControlCard component

Signed-off-by: Vikram Bala <vikrambala.va@gmail.com>

vickycore committed:
Updated CountCard component to fix sizing, added Profile Error card to results.vue, created ControlCard component

vickycore committed:
Merge branch 'dev' of https://github.com/mitre/heimdall-vue into vb/countcard

Signed-off-by: Vikram Bala <vikrambala.va@gmail.com>

vickycore committed:
Updated results.vue to include a fifth card for Profile Error.
Fixes #61 #62 #68 #69

Signed-off-by: Vikram Bala <vikrambala.va@gmail.com>

vickycore committed:
Merge branch 'vb/countcard' of https://github.com/mitre/heimdall-vue into vb/countcard

Signed-off-by: Vikram Bala <vikrambala.va@gmail.com>

Mitriol committed:
Merge pull request #83 from mitre/dev-typescript

Added typescript build support, and converted Store to reflect

Updated CountCard component to fix sizing, added Profile Error card to results.vue, created ControlCard component

Signed-off-by: Vikram Bala <vikrambala.va@gmail.com>

Updated results.vue to include a fifth card for Profile Error.
Fixes #61 #62 #68 #69

Signed-off-by: Vikram Bala <vikrambala.va@gmail.com>

vickycore committed:
Added multiple cards for status, severity, and compliance level and cleaned them up. Implemented filtering features.
Fixes #54, #72, #78, #79

Signed-off-by: Vikram Bala <vikrambala.va@gmail.com>

vickycore committed:
Added multiple cards for status, severity, and compliance level and cleaned them up. Implemented filtering features.
Fixes #54, #72, #78, #79

Signed-off-by: Vikram Bala <vikrambala.va@gmail.com>

lukemalinowski commented on src/components/CountCard.vue:
```diff
@@ -1,35 +1,38 @@
+        <check-circle-icon v-if="index==0" class="feathericon"></check-circle-icon>
+        <x-circle-icon v-else-if="index==1" class="feathericon"></x-circle-icon>
+        <slash-icon v-else-if="index==2" class="feathericon"></slash-icon>
+        <alert-triangle-icon v-else-if="index==3" class="feathericon"></alert-triangle-icon>
```
This is the kind of thing you'd do with a dirt simple functional component, if it was at all more complicated.

lukemalinowski replied:
By this i mean doing a selection from many html tags and icons based off a key :index in this case.

lukemalinowski commented on src/layouts/components/TheNavbar.vue:
```diff
@@ -345,6 +345,7 @@ export default {
       if (this.sidebarWidth == "default") return "navbar-default";
       else if (this.sidebarWidth == "reduced") return "navbar-reduced";
       else if (this.sidebarWidth) return "navbar-full";
+      return "";
```
This linter error is fixed in a future commit

vickycore committed:
Merge branch 'vb/countcard' of https://github.com/mitre/heimdall-vue into vb/countcard

Signed-off-by: Vikram Bala <vikrambala.va@gmail.com>

vickycore committed:
Merge vb/countcard branch with dev

Signed-off-by: Vikram Bala <vikrambala.va@gmail.com>

Commit:
ENH: Added proper linting support for typescript

Commit:
ENH: Added typescript compile to travis tests

Commit:
Removed typescript from production

FIX: Fixed package lock

Redid package lock

Commit:
Removed typescript from production

Commit:
FIX: Fixed package lock

aaronlippold committed:
fixed broken npm build
fixed proken package-lock.json file
added npm-merge-driver to the deps
added npm-merge-driver into the git repo and .gitattributes

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
Merge branch 'lint-typescript' of https://github.com/mitre/heimdall-vue into lint-typescript

aaronlippold committed:
fixed netlify redirect issue

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
removing last change, doesn't work

Signed-off-by: Aaron Lippold <lippold@gmail.com>

vickycore committed:
Fix sizing issues with charts and cards

Signed-off-by: Vikram Bala <vikrambala.va@gmail.com>

aaronlippold committed:
small fixes for sidebar and font to ensure function of main capability

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
Merge branch 'lint-typescript' into vb/countcard

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
updating package-lock.json

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
Merge branch 'dev' into vb/countcard

Signed-off-by: Aaron Lippold <lippold@gmail.com>
